plugin/storage/es/spanstore/writer.go

```diff
@@ -101,16 +101,12 @@ type spanAndServiceIndexFn func(spanTime time.Time) (string, string)
 func getSpanAndServiceIndexFn(p SpanWriterParams) spanAndServiceIndexFn {
 	spanIndexPrefix := p.IndexPrefix.Apply(spanIndexBaseName)
 	serviceIndexPrefix := p.IndexPrefix.Apply(serviceIndexBaseName)
-	if p.Archive {
-		return func(_ time.Time) (string, string) {
-			if p.UseReadWriteAliases {
-				return archiveIndex(spanIndexPrefix, archiveWriteIndexSuffix), ""
+	if p.UseReadWriteAliases {
+		if p.WriteAlias != "" {
+			return func(_ time.Time) (string, string) {
+				return spanIndexPrefix + p.WriteAlias, ""
 			}
-			return archiveIndex(spanIndexPrefix, archiveIndexSuffix), ""
```

Comment:
@yurishkuro do you know why the service index here is blank? is there any harm in returning this as well?

Reply:
I guess in the archive mode we never needed to write into service index since archive storage is only used to retrieve traces by ID, not to do any searches. Not sure if we can infer the same from some other signal / parameter.